File: 04-practice/ex03-words/solution.py
def unique_words(text):
    words = text.lower().split()
    cleaned = [w.strip(".,!?") for w in words if w.strip(".,!?")]
    return sorted(set(cleaned))


def assert_eq(actual, expected, label):
    ok = actual == expected
    print(f"OK   {label}" if ok else f"FAIL {label}: got {actual}")


assert_eq(
    unique_words("Hello, world! Hello Python."),
    ["hello", "python", "world"],
    "unique_words",
)

File: 04-practice/ex03-words/solution.js
function uniqueWords(text) {
  const parts = text.toLowerCase().split(/\s+/).filter(Boolean);
  const cleaned = parts.map((w) => w.replace(/^[.,!?]+|[.,!?]+$/g, ""));
  return [...new Set(cleaned)].sort();
}

function assertEq(actual, expected, label) {
  const ok = JSON.stringify(actual) === JSON.stringify(expected);
  console.log(ok ? `OK   ${label}` : `FAIL ${label}: got ${JSON.stringify(actual)}`);
}

assertEq(
  uniqueWords("Hello, world! Hello Python."),
  ["hello", "python", "world"],
  "uniqueWords"
);
